### PR TITLE
REV: NOS-17, Offliner should generate maven-metadata.xml files using downloaded files

### PR DESCRIPTION
**Pull request by yma88**
REV: NOS-17, Offliner should generate maven-metadata.xml files using downloaded files

### CODE REVIEW

**Comment by jdcasey** [comment]
This looks pretty good, but I have one question: Have you included the change away from scanning for POM paths as Offliner downloads, and toward scanning the filesystem afterward? It looks like it's tracking paths as they get downloaded.

Otherwise, the metadata generation step looks great.

**Comment by yma88** [comment]
John, what I now track the pom paths way is storing when it's verified as avoided and downloaded successfully, I think maybe some misunderstanding in the comment you replied to me in jira about this part which I've consult with you on my different thinkings (I originally think that if I could track the path which has been checked as avoided, plus successfull downloading ones, all situations I've handled, and no one else left, and this could replace the scanning the filesystem ) ...

**Comment by jdcasey** [comment]
Copying here for transparency:

Sorry, I read your earlier comment too quickly the first time around. I see what you're saying about capturing avoided and downloaded paths. It seems like only including avoided or downloaded paths should be fine, since they're all files downloaded for that manifest that are on the system. But what about other manifests that have been downloaded to the same directory?

The reason I was suggesting filesystem scanning is to basically write metadata for any files that were actually there, on the system. If the user ran Offliner multiple times with different manifest files, all pointing at the same directory, then files associated with early executions would be excluded from the metadata because they wouldn't be in the later manifest. I believe this violates the principle of least surprise...it would be surprising to have artifacts in the directory structure that the metadata files didn't know about.

So, I still think it might be simplest to just scan the directory recursively at the end instead of tracking paths avoided / downloaded during the execution. If you can think of a better way to include things that were downloaded during previous executions (with different manifest files) then of course we can look at that too.

Can we reopen this issue and make the switch to filesystem scanning or something similar?

File: src/main/java/com/redhat/rcm/offliner/Main.java [hunk 1]
```diff
@@ -183,6 +182,9 @@ else if ( result.isAvoided() )
                 }
             }
 
+            Set<String> pomPaths = new HashSet<>();
+            File download = new File( opts.getDownloads().getAbsolutePath() );
```
Comment (jdcasey) [inline src/main/java/com/redhat/rcm/offliner/Main.java]: I think this is already a File, so do we need to construct a new one? Or, if you need an absolute one, can you just use getAbsoluteFile()?

**Comment by jdcasey** [comment]
That last comment is very minor. The rest of this looks fine.

**Pull request merged**